allennlp_models/pair_classification/models/bimpm.py

```diff
@@ -246,3 +249,6 @@ def get_metrics(self, reset: bool = False) -> Dict[str, float]:
         }
 
     default_predictor = "textual_entailment"
+
+
+add_pretrained_model(id="pc-bimpm", name="Bilateral Multi-Perspective Matching Model")
```

Comment:
It seems odd to me to have these live in the model implementation classes, instead of in a central place that has all of the pretrained models.  Conceptually that means I'm modifying model code if I just want to add a new pretrained model, instead of modifying a central list of pretrained models.

Reply:
I also don't really understand what we gain by having a "pretrained model" added without any weights.  What is this doing?

Reply:
This is removed in the latest code commit. I was checking what models we have the pretrained archives for.

Reply:
Conceptually, when you change the model code, you should also re-train and update the trained weights.

It makes sense to me either way.